crates/wasmparser/src/readers/core/coredumps.rs

```diff
@@ -31,6 +31,94 @@ impl<'a> FromReader<'a> for CoreDumpSection<'a> {
     }
 }
 
+/// The data portion of a "coremodules" custom section. This contains a vec of
+/// module names that will be referenced by index by other coredump sections.
+///
+/// # Example
+///
+/// ```
+/// use wasmparser::{ BinaryReader, CoreDumpModulesSection, FromReader, Result };
+/// let data: &[u8] = &[0x01, 0x00, 0x04, 0x74, 0x65, 0x73, 0x74];
+/// let mut reader = BinaryReader::new(data);
+/// let modules_section = CoreDumpModulesSection::from_reader(&mut reader).unwrap();
+/// assert!(modules_section.modules[0] == "test")
+/// ```
+#[derive(Debug)]
+pub struct CoreDumpModulesSection<'a> {
+    /// A list of module names, which may be URLs, file paths, or other
+    /// identifiers for the module.
+    pub modules: Vec<&'a str>,
+}
+
+impl<'a> FromReader<'a> for CoreDumpModulesSection<'a> {
+    fn from_reader(reader: &mut BinaryReader<'a>) -> Result<Self> {
+        let pos = reader.original_position();
+        let mut modules = vec![];
+        for _ in 0..reader.read_var_u32()? {
+            if reader.read_u8()? != 0 {
+                bail!(pos, "invalid start byte for coremodule");
+            }
+            modules.push(reader.read_string()?);
+        }
+        Ok(CoreDumpModulesSection { modules })
+    }
+}
+/// A custom section representing the instances involved in a given coredump
+pub struct CoreDumpInstancesSection {
+    /// The instances for the coredump
+    pub instances: Vec<CoreDumpInstance>,
+}
+
+impl<'a> FromReader<'a> for CoreDumpInstancesSection {
+    fn from_reader(reader: &mut BinaryReader<'a>) -> Result<Self> {
+        let mut instances = vec![];
+        for _ in 0..reader.read_var_u32()? {
+            instances.push(CoreDumpInstance::from_reader(reader)?);
+        }
+        Ok(CoreDumpInstancesSection { instances })
+    }
+}
+
+/// A single instance from a coredump instances section
+pub struct CoreDumpInstance {
+    /// The module that this is an instance of, as an index into a "coremodules"
+    /// section.
+    pub module_index: u32,
+
+    /// Which of the coredump's memories are this instance's memories, via
+    /// indexing into the memory index space.
+    pub memories: Vec<u32>,
+
+    /// Which of the coredump's globals are this instance's globals, via
+    /// indexing into the global index space.
+    pub globals: Vec<u32>,
```

Comment:
All of these `Vec`s we will probably want to turn into sub-readers eventually, but we can wait till we have a reason to do that.